Obsidian/WorldData/Generators/Overworld/Features/Trees/BaseTree.cs

```diff
@@ -94,7 +94,7 @@ protected virtual async Task GenerateLeavesAsync(Vector origin, int heightOffset
                         chunk);
 
                     if (xx == 0 || zz == 0)
-                        await helper.SetBlockAsync(origin.X + xx, trunkHeight + origin.Y + 2, origin.Z + zz,
+                        await helper.SetBlockAsync(origin.X + xx, trunkHeight + origin.Y + heightOffset, origin.Z + zz,
```

Comment:
Surely it's the case that the other leaf blocks need to account for `heightOffset` too, right?